SimCalorimetry/HGCalSimProducers/plugins/alpaka/HGCalDigiSoaFiller.cc

```diff
@@ -0,0 +1,185 @@
+    desc.add<edm::InputTag>("cehsiChannelDigis", edm::InputTag("simHGCalUnsuppressedDigis", "HEfront"));
+    desc.add<edm::InputTag>("cehsipmChannelDigis", edm::InputTag("simHGCalUnsuppressedDigis", "HEback"));
+    desc.add<std::string>("digisProdName", "HGC");
+    descriptions.addWithDefaultLabel(desc);
+  }
+
+  //
+  void HGCalDigiSoaFiller::produce(device::Event& iEvent, device::EventSetup const& iSetup) {
+    cfgWatcher_.check(iSetup);
+    //const CellIndexer &cellidx = iSetup.getData(cellIndexTkn_);
+    const CellInfo& cells = iSetup.getData(cellTkn_);
+    const ModuleIndexer& modidx = iSetup.getData(moduleIndexTkn_);
+    const ModuleInfo& modules = iSetup.getData(moduleTkn_);
+
+    //FIXME
+    int itSample = 2;
+
+    //read "classic" channel digis
+    auto const& digisCEE = iEvent.get(digisCEET_);
+    //auto const &digisCEHSi = iEvent.get(digisCEHSiT_);
+    //auto const &digisCEHSiPM = iEvent.get(digisCEHSiPMT_);
+
+    //fill the SoA collection in the host
+    uint32_t finaldigi_size = modidx.maxDataIdx_;
+    std::cout << "Allocating SOA with " << finaldigi_size << " entries" << std::endl;
+    hgcaldigi::HGCalDigiHost host_buffer(finaldigi_size, cms::alpakatools::host());
+    int nmatch(0), nfail(0);
+    for (auto d : digisCEE) {
+      //det id to electronics id
+      uint32_t detid(d.id().rawId());
+      HGCalElectronicsId eleid(
+          ::hgcal::mappingtools::getElectronicsIdForSiCell<ModuleInfo, CellInfo>(modules, cells, detid));
+      if (eleid.raw() == 0) {
+        nfail++;
+        continue;
+      }
+
+      //get the index in the SOA
+      auto modcellidx =
+          ::hgcal::mappingtools::getModuleCellIndicesForSiCell<ModuleInfo, CellInfo>(modules, cells, detid);
+      uint32_t i = finaldigi_size;
+      try {
+        i = modidx.getIndexForModuleData(
+            eleid.localFEDId(), eleid.captureBlock(), eleid.econdIdx(), eleid.econdeRx(), eleid.halfrocChannel());
+        if (i >= finaldigi_size) {
+          HGCSiliconDetId siid(d.id());
+          int32_t midx = modcellidx.first;
+          int32_t cidx = modcellidx.second;
+          std::cout << "Failed to get proper index (" << i << ") for "
+                    << "layer=" << siid.layer() << " u=" << siid.waferU() << " v=" << siid.waferV() << std::endl
+                    << "\t Modidx:" << midx << " cellidx: " << cidx << " cellu=" << siid.cellU()
+                    << " cellv=" << siid.cellV() << std::endl
+                    << "\tModule info FED:" << (uint32_t)modules.view()[midx].fedid()
+                    << " CB:" << (uint32_t)modules.view()[midx].captureblockidx()
+                    << " ECOND: " << (uint32_t)modules.view()[midx].econdidx()
+                    << "\tEle id: FED=" << (uint32_t)eleid.localFEDId() << " CB:" << (uint32_t)eleid.captureBlock()
+                    << " ECONDidx: " << (uint32_t)eleid.econdIdx() << std::endl;
+          nfail++;
+          continue;
+        }
+      } catch (std::exception& e) {
```

Comment:
Could you describe the exception case here? What component is expected to throw an exception, and in what kind of situation?

Reply:
This file was remove all together in https://github.com/CMS-HGCAL/cmssw/pull/137.